### ISSUE TITLE
Server Error 503 (Max Time/Memory) in v2.9.1

### ISSUE THREAD

**Issue by wpsmort** [semperfiwebdesign/all-in-one-seo-pack#2008]
This is related to #1997.

Whilst testing PR #2007 I was able to reproduce the fatal error below reliably on a clone of a client site.   The site has nearly 8,000 posts and was using static XML sitemaps.  When I installed AIOSEOP v2.9 the site crashed with a 503 error and these errors:

PHP Fatal error:  Allowed memory size of 268435456 bytes exhausted (tried to allocate 72 bytes) in /wp-content/plugins/all-in-one-seo-pack-1997/modules/aioseop_sitemap.php on line 3089

WordPress database error Lost connection to MySQL server during query for query SELECT * FROM hon_posts WHERE ID = 27621 LIMIT 1 made by do_action('admin_init'), WP_Hook->do_action, WP_Hook->apply_filters, call_user_func_array, AIOSEOP_Updates->version_updates, do_action('after_doing_aioseop_updates'), WP_Hook->do_action, WP_Hook->apply_filters, call_user_func_array, All_in_One_SEO_Pack_Sitemap->do_sitemaps, All_in_One_SEO_Pack_Sitemap->do_simple_sitemap, All_in_One_SEO_Pack_Sitemap->get_simple_sitemap, All_in_One_SEO_Pack_Sitemap->get_all_post_priority_data, All_in_One_SEO_Pack_Sitemap->get_prio_from_posts, All_in_One_SEO_Pack_Sitemap->get_images_from_post, wp_get_attachment_image_url, wp_get_attachment_image_src, image_downsize, wp_attachment_is_image, wp_attachment_is, get_post, WP_Post::get_instance

WordPress database error Lost connection to MySQL server during query for query SELECT post_id, meta_key, meta_value FROM hon_postmeta WHERE post_id IN (27625) ORDER BY meta_id ASC made by require_once('wp-admin/admin.php'), do_action('admin_init'), WP_Hook->do_action, WP_Hook->apply_filters, call_user_func_array, AIOSEOP_Updates->version_updates, do_action('after_doing_aioseop_updates'), WP_Hook->do_action, WP_Hook->apply_filters, call_user_func_array, All_in_One_SEO_Pack_Sitemap->do_sitemaps, All_in_One_SEO_Pack_Sitemap->do_simple_sitemap, All_in_One_SEO_Pack_Sitemap->get_simple_sitemap, All_in_One_SEO_Pack_Sitemap->get_all_post_priority_data, All_in_One_SEO_Pack_Sitemap->get_prio_from_posts, All_in_One_SEO_Pack_Sitemap->get_images_from_post, wp_get_attachment_image_url, wp_get_attachment_image_src, image_downsize, wp_attachment_is_image, wp_attachment_is, get_attached_file, get_post_meta, get_metadata, update_meta_cache

This is not new.  I was able to reproduce this with v2.8 by switching from dynamic to static sitemap.  When I did this the site crashed with the same error.  

This could easily be resolved in two ways:

1. Stop creating static sitemaps.  This has always been our goal.
2. Enable sitemap indexes by default.

**Comment by wpsmort** [comment]
Contact me on Slack for the access to the Cloudways site where we can reproduce this.

**Comment by contactashish13** [comment]
NOTE: When we remove static sitemap, test cases might get impacted as that feature is being used in the test cases.

**Comment by EkoJR** [comment]
This is partially related to #1997 which was to fix this (different version though).

`get_image_attributes` was the other function that was being spotted at the time, and is still showing up on xDebug Profile (using QCacheGrind).

**Comment by wpsmort** [comment]
Reported again by Mark Zetter by email.  He sent this from his hosting company LiquidWeb:

The basis of my hypothesis is the system calls that PHP is doing at the time of the long delays. It is sending MySQL queries like this one:
 
9:02:55.905636 sendto(4<UNIX:[591731->593118]>, "s\0\0\0\3SELECT ID FROM wp97stgd_posts WHERE guid='http
s://www.ventureoutsource.com/wp-content/uploads/2017/05/Leesys.png';", 119, MSG_DONTWAIT, NULL, 0) = 119
19:02:55.905719 poll([{fd=4<UNIX:[591731->593118]>, events=POLLIN|POLLERR|POLLHUP}], 1, 1471228928) = 1 (
[{fd=4, revents=POLLIN}])

System calls look fairly gross so here is that query cleaned up:
SELECT ID FROM wp97stgd_posts WHERE guid='https://www.ventureoutsource.com/wp-content/uploads/2017/05/Leesys.png';

I used wildcards to search for a query like that in every PHP file in your site and there was 1 result:
 
[root@host public_html]# find . -name '*.php' | sed 's/ /\\ /g' | xargs grep 'SELECT ID FROM.*posts.*WHERE.*guid=' * -sl
wp-content/plugins/all-in-one-seo-pack/modules/aioseop_sitemap.php

If you have any questions just let me know.

**Comment by wpsmort** [comment]
Reported again here - https://wordpress.org/support/topic/all-in-one-how-to-deactivate-admin-panel-blank/

I got access to this site and found that he has 8515 posts, 2545 events and 250406 media items.  Dynamic Sitemap was enabled but Sitemap Indexes was not.  Yst SEO was also running.  

The error log showed these fatal errors:

Allowed memory size of 268435456 bytes exhausted (tried to allocate 71 bytes) in /wp-includes/wp-db.php on line 2385, referer: /wp-admin/admin.php?page=all-in-one-seo-pack%2Fmodules%2Faioseop_performance.php

Allowed memory size of 268435456 bytes exhausted (tried to allocate 71 bytes) in /wp-includes/wp-db.php on line 2385, referer: /wp-admin/admin.php?page=all-in-one-seo-pack%2Fmodules%2Faioseop_feature_manager.php

I installed v2.8 and activated Sitemap Indexes and pagination to temporarily resolve the problem.

**Comment by EkoJR** [comment]
With dynamic sitemaps, the problem only occurs on the sitemap.
With static sitemaps, the problem occurs when saving settings and when content is updated.

**Comment by wpsmort** [comment]
For many of the sites I've gotten access to, the problem occurs upon plugin activation.  At that point the WP Admin becomes inaccessible.  On the latest site I got access to for Phillip Berry, I had to roll back to v2.9 of Pro and reduce sitemap pagination from 50000 (the default setting) to 1000.  I could then get the latest release of Pro to work but the site was very slow on all page loads in WP Admin.  

We should look at whether the addition of the RSS Sitemap feature may be a possible cause for this.  I feel that RSS Sitemap feature should really be an option and should be disabled by default.
